src/core/AudioEngine.cpp

```diff
@@ -481,24 +479,41 @@ void AudioEngine::handleMetronome()
 	}
 
 	tick_t ticks = song->getPlayPos(currentPlayMode).getTicks();
-	tick_t ticksPerBar = TimePos::ticksPerBar();
-	int numerator = song->getTimeSigModel().getNumerator();
-
-	if (ticks == lastMetroTicks)
+	if (ticks == m_prevMetronomeStates.second)
 	{
 		return;
 	}
 
-	if (ticks % (ticksPerBar / 1) == 0)
+	/* Metronome sound generation - explanation:
+	 * The modulo operation below figures out the exact point in time when a sound should be played. As this operation
+	 * relies on ".getTicks()" we have to add a certain tick tolerance, since it is not guaranted that between each call
+	 * the tick gets a +1-increment and thus the exact point in time can be missed.*/	
+
+	// increase this parameter if with complex projects or poor HW resources the metronome rings out of rythm
+	int nToleranceTicks = 5;
+	// LMMS allows abrupt jumps in song positions (e.g. song-loops, when marker was put of that loop), this iscached below
+	bool noDoubleTick = m_prevMetronomeStates.first > nToleranceTicks;
+	// identify which sound should be played below
+	bool highTickEvent = ticks % int(TimePos::ticksPerBar() / 1) <= nToleranceTicks;
+	bool lowTickEvent = ticks % int(TimePos::ticksPerBar() / song->getTimeSigModel().getNumerator()) <= nToleranceTicks;
+
+	// handle sound-'state'-machine
```

Comment:
If this is a 'state' machine we could probably change this to a `switch` instead. I think it would allow easier future additions, possibly adding in more types of ticks to play.

Reply:
Are there any comparable benefits for a `switch` ? I'm not sure if the necessary extra lines that enable this structure in that context are an advantage here.

Also, on another PR ( #6119 ) I'm picking this up actually, but it is currently on hold on until this one is clarified